intel-retail/automated-self-checkout
# Add model ready waiting logic to resolve issue #331

- Add gRPC calls in profile-launcher to test models readiness from ovms server
 - Refactor how the unit-test makefile runs as now it requires google gRPC packages

## PR Checklist



## What are you changing?


## Issue this PR will close

close: #**331**

## Anything the reviewer should know when reviewing this PR?

## Test Instructions if applicable

- git clone this PR
- make build-profile-launcher
- make clean-all
- make run-camera-simulator
- run multiple pipelines, say, classification, for more than 5 pipelines and observe that the 6th nd 7th pipelines can also be launched without problem now:
```console
$ PIPELINE_PROFILE="classification" RENDER_MODE=0 sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0

```
- run unit tests:  `make unit-test-profile-launcher`  should see ALL PASSED ok
- run smoke tests: `make run-smoke-tests` should see ALL PASSED


## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/automated-self-checkout )
